core/src/data/rasterSource.cpp

```diff
@@ -2,58 +2,78 @@
 #include "data/propertyItem.h"
 #include "data/tileData.h"
 #include "tile/tile.h"
+#include "tile/tileBuilder.h"
 #include "tile/tileTask.h"
 #include "util/mapProjection.h"
-#include "platform.h"
+#include "log.h"
 
 namespace Tangram {
 
 class RasterTileTask : public BinaryTileTask {
 public:
-    RasterTileTask(TileID& _tileId, std::shared_ptr<TileSource> _source, int _subTask)
-        : BinaryTileTask(_tileId, _source, _subTask) {}
 
-    std::shared_ptr<Texture> m_texture;
+    const bool subTask = false;
+
+    std::unique_ptr<Texture> texture;
+    std::unique_ptr<Raster> raster;
+
+    RasterTileTask(TileID& _tileId, std::shared_ptr<TileSource> _source, bool _subTask)
+        : BinaryTileTask(_tileId, _source),
+          subTask(_subTask) {}
+
 
     std::shared_ptr<RasterSource> rasterSource() {
         return reinterpret_cast<std::weak_ptr<RasterSource>*>(&m_source)->lock();
     }
 
     bool hasData() const override {
-        return bool(rawTileData) || bool(m_texture);
+        return bool(rawTileData) || bool(texture) || bool(raster);
     }
 
     bool isReady() const override {
-        if (!isSubTask()) {
+        if (!subTask) {
             return bool(m_tile);
         } else {
-            return bool(m_texture);
+            return bool(texture) || bool(raster);
         }
     }
 
     void process(TileBuilder& _tileBuilder) override {
         auto source = rasterSource();
         if (!source) { return; }
 
-        if (!m_texture) {
+        if (!texture && !raster) {
             // Decode texture data
-            m_texture = source->createTexture(m_tileId, *rawTileData);
+            texture = source->createTexture(m_tileId, *rawTileData);
+            if (!texture) {
+                raster = std::make_unique<Raster>(m_tileId, source->emptyTexture());
+            }
         }
 
         // Create tile geometries
-        if (!isSubTask()) {
-            BinaryTileTask::process(_tileBuilder);
+        if (!subTask) {
+            m_tile = _tileBuilder.build(m_tileId, *(source->m_tileData), *source);
```

Comment:
I am not sure this is correct, when this task is coming from a primary "RasterSource", we will need to call `process` to get tile data.

Reply:
Probably this is the reason for the crash @matteblair noticed.

Reply:
Look at `TileTask::process`: it calls parse to get the `TileData` then runs `tileBuilder.builder()` with it. RasterSource provides always the same `TiledData` for a simple rectangle mesh - so we can skip the `TileSource::parse()` indirection here.
`source->m_tileData` should valid when RasterSource canGenerateGeometry.... ok hold on I know where the bug is

Reply:
Agreed on the redirection, and looks like you fixed this specific bug here: ed544d6.
ed544d6, does fix the tiledata being null here.